common/planners/src/plan_stage.rs

```diff
@@ -6,27 +6,29 @@ use std::sync::Arc;
 
 use common_datavalues::DataSchemaRef;
 
+use crate::Expression;
 use crate::PlanNode;
 
-#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
-pub enum StageState {
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub enum StageKind {
     Normal,
-    Through,
-    SortMerge,
-    GroupByMerge,
-    AggregatorMerge
+    Expansive,
+    Convergent
```

Comment:
How about the plan that preserves the cardinality?

Reply:
What is `preserves the cardinality`?